runtime/src/iree/builtins/ukernel/common.h

```diff
@@ -428,6 +428,20 @@ static inline iree_uk_type_t iree_uk_unpack_type(int pos,
 #define IREE_UK_ASSUME_UNREACHABLE
 #endif
 
+// Queries for [[attribute]] identifiers in modern compilers.
+#ifdef __has_attribute
+#define IREE_UK_HAVE_ATTRIBUTE(x) __has_attribute(x)
+#else
+#define IREE_UK_HAVE_ATTRIBUTE(x) 0
+#endif  // __has_attribute
+
+#if IREE_UK_HAVE_ATTRIBUTE(noinline) || \
+    (defined(__GNUC__) && !defined(__clang__))
```

Comment:
From some Googling, `__has_attribute` is supported since at least Clang 3.4 and GCC 5. So no need for anything else as far as we're concerned.

In the above implementation of `IREE_UK_HAS_BUILTIN`, I included the `__GNUC__` check because `__has_builtin` is only supported in GCC 10.  Even so, this seemed close to good enough for us, except perhaps on some build bot...

Reply:
(Nevermind the suggestions since auto-merged kicked in; I'll fold these changes in another PR).

Reply:
also taken from iree/base/attributes.h - if you think this can be simplified it'd be good to do it in both places